xray/plot/plot.py

```diff
@@ -456,11 +472,16 @@ def newplotfunc(darray, ax=None, xincrease=None, yincrease=None,
                                  vmax=cmap_params['vmax'],
                                  **kwargs)
 
-        ax.set_xlabel(xlab)
-        ax.set_ylabel(ylab)
+        # Label the plot with metadata
+        if add_labels:
+            ax.set_xlabel(xlab)
+            ax.set_ylabel(ylab)
+            ax.set_title(_title_for_slice(darray))
 
         if add_colorbar:
-            plt.colorbar(primitive, ax=ax, extend=cmap_params['extend'])
+            cbar = plt.colorbar(primitive, ax=ax, extend=cmap_params['extend'])
+            if darray.name and add_labels:
```

Comment:
This was really a nitpicky comment. The only sure way to know the user hasn't set the name attribute to something is to compare it to `None`.  For example, `False` and `0` both evaluate to `False` so if the user, for some reason, set their DataArray to one of those, the current syntax would fail to add the label: 

```python
In [9]: a.name = 0

In [10]: if a.name:
   ....:     print('fail')

In [11]:
```

Maybe not the worst thing, but *could* lead to some confusion at some point.